app/controllers/users_controller.rb

```diff
@@ -1,10 +1,8 @@
 class UsersController < ApplicationController
 
-  skip_before_filter :registration,
-                     only: [:ask_for_email, :edit, :update]
+  skip_before_filter :registration, only: [:ask_for_email]
```

Comment:
I *think* that this controller action is only for the edit profile page. The registration uses a different one.